services/Postmates.mjs

```diff
@@ -280,5 +282,311 @@ class Postmates extends Service {
       responseCookies: this.cookiesToJson(res.headers.raw()["set-cookie"]),
     };
   }
+
+  /*Initialize Postmates cart.
+   * @params itemID->itemName are obtained from getStore() and getItemDetails().
+   * @param {itemQuantity} Number of the item to add to the cart.
+   * @param cookies {Object} cookie to pass in location data.
+   * @return {Object} Raw cart information response.
+   */
+  async createCart(
+    {
+      itemID,
+      storeID,
+      sectionID,
+      subsectionID,
+      priceAsCents,
+      itemName,
+      itemQuantity,
+      customizations,
+      imageURL,
+    },
+    cookies
+  ) {
+    const generatedItemUUID = uuidv4();
+    const res = await fetch("https://postmates.com/api/createDraftOrderV2", {
+      method: "POST",
+      headers: {
+        authority: "postmates.com",
+        accept: "*/*",
+        "accept-language": "en-US,en;q=0.9",
+        "content-type": "application/json",
+        cookie: this.CookiesObjectToString(cookies),
+        dnt: "1",
+        origin: "https://postmates.com",
+        "sec-ch-ua":
+          '"Chromium";v="110", "Not A(Brand";v="24", "Google Chrome";v="110"',
+        "sec-ch-ua-mobile": "?0",
+        "sec-ch-ua-platform": '"Windows"',
+        "sec-fetch-dest": "empty",
+        "sec-fetch-mode": "cors",
+        "sec-fetch-site": "same-origin",
+        "user-agent":
+          "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36",
+        "x-csrf-token": "x",
+      },
+      body: JSON.stringify({
+        isMulticart: false,
+        shoppingCartItems: [
+          {
+            uuid: itemID,
+            shoppingCartItemUuid: generatedItemUUID,
+            storeUuid: storeID,
+            sectionUuid: sectionID,
+            subsectionUuid: subsectionID,
+            price: priceAsCents,
+            title: itemName,
+            quantity: itemQuantity,
+            customizations: customizations,
+            imageURL: imageURL,
+            specialInstructions: "",
+            itemId: null,
+          },
+        ],
+        useCredits: true,
+        extraPaymentProfiles: [],
+        promotionOptions: {
+          autoApplyPromotionUUIDs: [],
+          selectedPromotionInstanceUUIDs: [],
+          skipApplyingPromotion: false,
+        },
+        deliveryTime: {
+          asap: true,
+        },
+        deliveryType: "ASAP",
+        currencyCode: "USD",
+        interactionType: "door_to_door",
+        checkMultipleDraftOrdersCap: false,
+        isGuestOrder: true,
+        businessDetails: {
+          profileType: "personal",
+        },
+      }),
+    });
+    if (res.ok) {
+      return {
+        data: await res.json(),
+        responseCookies: this.cookiesToJson(res.headers.raw()["set-cookie"]),
+      };
+    } else {
+      throw new HTTPResponseError(res);
```

Comment:
Notice how this error keeps getting caught everywhere, instead of repeating this code, lets use the `this.callServiceAPI()` method. It takes care of catching the errors. You can find examples of how to use this method in other methods of this class.

Reply:
Fixed I think